msu/hooks/skills/skill.nut

```diff
@@ -359,24 +357,24 @@
 
 		container.onBeforeAnySkillExecuted(this, _targetTile, targetEntity, _forFree);
 
-		local ret = use(_targetTile, _forFree);
+		local ret = __original(_targetTile, _forFree);
 
 		container.onAnySkillExecuted(this, _targetTile, targetEntity, _forFree);
 
 		return ret;
 	}
 
-	o.getDamageType <- function()
+	q.getDamageType <- function()
 	{
 		return this.m.DamageType;
 	}
 
-	o.getWeightedRandomDamageType <- function()
+	q.getWeightedRandomDamageType <- function()
 	{
-		return this.m.DamageType.roll();
+		return this.m.DamageType == null ? null : this.m.DamageType.roll();
 	}
```

Comment:
This seems like a bugfix

Reply:
This bug only occurs as a result of porting to Modern Hooks, so yeah it needs a fix only here :P. Because now we initialize `this.m.DamageType` to `null` by default because of the cloning issue we discussed.

Reply:
I see